core/src/main/java/com/cyberbot/bomberman/core/models/tiles/TileFactory.java

```diff
@@ -1,55 +1,71 @@
-     * @param world Box2D world reference, required for any {@link PhysicalTile PhysicalTiles}.
-     * @param x     The x component of the tile's position.
-     * @param y     The y component of the tile's position.
-     * @return A new instance of a Tile.
-     * @throws InvalidPropertiesFormatException When some required properties where missing from the TiledMapTile
-     *                                          or were of an invalid type.
-     * @throws IllegalArgumentException         When a property contains an illegal value.
-     */
-    public static Tile createTile(TiledMapTile tile, World world, int x, int y)
-        throws InvalidPropertiesFormatException {
-        MapProperties properties = tile.getProperties();
-        if (!properties.containsKey(Tile.PROPERTY_TILE_TYPE) ||
-            !properties.containsKey(Tile.PROPERTY_TEXTURE_NAME)) {
+
+
+    private static final String PROPERTY_TILE_TYPE = "tile_type";
+    private static final String PROPERTY_TEXTURE_NAME = "texture";
+
+    private static final String TILE_TYPE_FLOOR = "floor";
+    private static final String TILE_TYPE_WALL = "wall";
+    private static final String TILE_TYPE_BASE = "base";
+
+    public static Tile createTile(NodeList tile, World world, int x, int y) throws InvalidPropertiesFormatException {
+        if (tile == null) {
+            return null;
+        }
+
+        HashMap<String, Object> properties = new HashMap<>();
+
+        for (int i = 0; i < tile.getLength(); i++) {
+            Element element = (Element) tile.item(i);
+            switch (element.getAttribute("type")) {
+                case "float":
+                    properties.put(element.getAttribute("name"),
+                        Float.parseFloat(element.getAttribute("value")));
+                    break;
+                case "int":
+                    properties.put(element.getAttribute("name"),
+                        Integer.parseInt(element.getAttribute("value")));
+                    break;
+                default:
+                    properties.put(element.getAttribute("name"), element.getAttribute("value"));
+            }
+        }
+        if (!properties.containsKey(PROPERTY_TILE_TYPE) || !properties.containsKey(PROPERTY_TEXTURE_NAME)) {
             throw new InvalidPropertiesFormatException(
                 "Each tile in the tile map has to contain '" +
-                    Tile.PROPERTY_TILE_TYPE + "' and '" +
-                    Tile.PROPERTY_TEXTURE_NAME + "' properties"
+                    PROPERTY_TILE_TYPE + "' and '" +
+                    PROPERTY_TEXTURE_NAME + "' properties"
             );
         }
 
-        String tileType = properties.get(Tile.PROPERTY_TILE_TYPE, String.class);
-        String textureName = properties.get(Tile.PROPERTY_TEXTURE_NAME, String.class);
+        String tileType = (String) properties.get(PROPERTY_TILE_TYPE);
+        String textureName = (String) properties.get(PROPERTY_TEXTURE_NAME);
```

Comment:
Should catch `ClassCastException` and throw `InvalidPropertiesFormatException`

Reply:
Fixed